controllers/humiocluster_controller.go

```diff
@@ -1922,6 +1928,64 @@ func (r *HumioClusterReconciler) ensureMismatchedPodsAreDeleted(ctx context.Cont
 
 	podsForDeletion := desiredLifecycleState.podsToBeReplaced
 
+	pdbSpec := hnp.GetPodDisruptionBudget()
+	r.Log.Info("Entering PDB enforcement check", "nodePool", hnp.GetNodePoolName(), "pdbSpec", pdbSpec, "pdbSpec.Enabled", pdbSpec != nil && pdbSpec.Enabled)
+	if pdbSpec != nil && pdbSpec.Enabled {
+		pdbName := fmt.Sprintf("%s-%s-pdb", hc.Name, hnp.GetNodePoolName())
+		pdb := &policyv1.PodDisruptionBudget{}
+
+		r.Log.Info("Fetching PDB",
+			"pdbName", pdbName,
+			"namespace", hnp.GetNamespace())
+
+		err := r.Get(ctx, types.NamespacedName{
+			Name:      pdbName,
+			Namespace: hnp.GetNamespace(),
+		}, pdb)
+
+		if err != nil {
+			if k8serrors.IsNotFound(err) {
+				r.Log.Info("PDB not found for node pool, proceeding without PDB check",
+					"pdb", pdbName,
+					"namespace", hnp.GetNamespace(),
```

Comment:
The same thing is true here where you don't need to explicitly add namespace to the log events, since the logger already includes this to all events.

Reply:
Same as above